src/main/java/io/github/cottonmc/resources/MetalResourceType.java

```diff
@@ -0,0 +1,82 @@
+package io.github.cottonmc.resources;
+
+import java.util.function.Supplier;
+
+import io.github.cottonmc.cotton.registry.CommonBlocks;
+import net.minecraft.block.Block;
+import net.minecraft.item.Item;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+public class MetalResourceType extends ItemResourceType {
+    protected Supplier<Block> oreSupplier = BlockSuppliers.STONE_TIER_ORE;
+
+    public MetalResourceType(String name) {
+        super(name);
+        withItemAffixes("ingot", "nugget", "dust");
+    }
+
+    public MetalResourceType withItemAffixes(String... affixes) {
+        for(String affix : affixes) this.affixes.add(affix);
+        return this;
+    }
+
+    public MetalResourceType withOreSupplier(Supplier<Block> oreSupplier) {
+        this.oreSupplier = oreSupplier;
+        return this;
+    }
+
+    @Override
+    public String getDomain() {
+        return name;
+    }
+
+    @Override
+    public boolean governs(String itemName) {
+        if (super.governs(itemName)) return true;
+        
+        //blocks
+        if ((name+"_ore").equals(itemName)) return (oreSupplier!=null);
+        if ((name+"_block").equals(itemName)) return true;
+        return false;
+    }
+
+    @Override
+    public Item getItem(String itemName) {
+        if (itemName.equals(name+"_ore") || itemName.equals(name+"_block")) {
```

Comment:
Don't hardcode the available blocks here.

Reply:
This is the metal provider, where metal storage blocks are *extremely* non-controversial, and ore blocks are fairly controversial, each needing a special provider (or no provider). I'd definitely like to see a more flexible/generalized one, but it kind of made sense to me to hardcode here.

Reply:
My problem is that you also hardcode it at getBlock and in registerAllBlocks. Maybe add separate Affixes for blocks?

Reply:
done.